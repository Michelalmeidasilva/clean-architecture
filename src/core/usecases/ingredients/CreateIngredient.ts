import { Ingredient } from "../../entity/ingredients/Ingredient";
 
export class CreateIngredient {
  invoke({ unit, title, id }: Ingredient): Ingredient {
    return {
      unit,
      title,
      id,
    };
  }
}
