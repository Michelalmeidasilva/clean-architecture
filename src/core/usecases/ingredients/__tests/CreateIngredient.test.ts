import { test, expect, describe } from "bun:test";

import { CreateIngredient } from "../CreateIngredient";
import { UnitMeasure } from "@core/entity/index";

const IngredientMock = {
  title: "FeijÃ£o",
  unit: UnitMeasure.LT,
  id: "hash_3424234",
};

describe("CreateIngredient", () => {
  test("Should Create a new ingredient", () => {
    const ingredient = new CreateIngredient();

    console.log(ingredient);

    expect(ingredient).toMatchObject(IngredientMock);
  });
});
