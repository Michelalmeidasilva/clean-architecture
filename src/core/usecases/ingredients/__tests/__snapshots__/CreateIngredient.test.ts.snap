// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIngredient Should Create a new ingredient 1`] = `
{
  "id": "hash_3424234",
  "title": "Feij√£o",
  "unit": "lt",
}
`;
